tests/ballerina-spec-conformance-tests/src/test/java/org/ballerinalang/test/TestRunnerUtils.java

```diff
@@ -0,0 +1,581 @@
+        for (int i = 0; i < noOfDiagnostics; i++) {
+            String actualLineNum = actualLineNumbers.get(i);
+            String expLineNum;
+            String expValue;
+            try {
+                expLineNum = String.valueOf(lineNumbers.get(i) + absLineNum);
+                expValue = outputValues.get(i);
+            } catch (IndexOutOfBoundsException e) {
+                expLineNum = null;
+                expValue = null;
+            }
+            setResultsAttributes(context, actualErrorMessages.get(i), expValue, actualLineNum, expLineNum);
+            Assert.assertEquals(actualLineNum, expLineNum);
+        }
+        if (outputValues.size() > noOfDiagnostics) {
+            setResultsAttributes(context, null, outputValues.get(noOfDiagnostics), null,
+                                 String.valueOf(lineNumbers.get(noOfDiagnostics)));
+            Assert.assertNull(iterator.next().toString());
+        }
+    }
+
+    public static void validateOutput(CompileResult compileResult, List<String> outputValues, List<Integer> lineNumbers,
+                                      int absLineNum, ITestContext context) {
+        Collection<Diagnostic> diagnostics = compileResult.getDiagnosticResult().errors();
+        if (!diagnostics.isEmpty()) {
+            Diagnostic diagnostic = diagnostics.iterator().next();
+            String message = diagnostic.message().replace(CARRIAGE_RETURN_CHAR, EMPTY_STRING);
+            String actualLineNum =
+                   String.valueOf(diagnostic.location().lineRange().startLine().line() + absLineNum);
+            setResultsAttributes(context, message, null, actualLineNum,
+                                 String.valueOf(lineNumbers.get(0) + absLineNum));
+            Assert.assertNull(diagnostic.toString());
+        }
+
+        BRunUtil.ExitDetails exitDetails = BRunUtil.run(compileResult);
+        if (!exitDetails.errorOutput.isEmpty()) {
+            String outputVal = outputValues.get(0);
+            setResultsAttributes(context, exitDetails.errorOutput, outputVal, null,
+                                 String.valueOf(lineNumbers.get(0) + absLineNum));
+            Assert.assertEquals(outputVal, exitDetails.errorOutput);
+        }
+
+        String consoleOutput = exitDetails.consoleOutput;
+        String[] results = consoleOutput.split("(\r\n|\r|\n)", -1);
+        int size = outputValues.size();
+        for (int i = 0; i < size; i++) {
+            String expectedOutput = outputValues.get(i);
+            String actualOutput;
+            try {
+                actualOutput = results[i];
+            } catch (ArrayIndexOutOfBoundsException e) {
+                actualOutput = null;
+            }
+            String lineNo = String.valueOf(lineNumbers.get(i) + absLineNum);
+            setResultsAttributes(context, actualOutput, expectedOutput, lineNo, lineNo);
+            Assert.assertEquals(expectedOutput, actualOutput);
```

Comment:
Still incorrect. Actual should be the [first argument](https://www.javadoc.io/doc/org.testng/testng/6.8.17/org/testng/Assert.html#assertEquals(java.lang.String,%20java.lang.String)).

Please check wherever applicable.